check.go

```diff
@@ -36,26 +49,63 @@ func check(pkg *gx.Package) (bool, error) {
 			return nil
 		})
 	}
+
 	if err := traverse(pkg); err != nil {
 		return !failed, err
 	}
+
+	var dupes []string
 	for name, pkgVersions := range packages {
 		switch len(pkgVersions) {
 		case 0:
 			panic("must have at least one package version")
 		case 1:
 			continue
 		}
+		dupes = append(dupes, name)
+	}
+
+	if len(dupes) > 0 {
 		failed = true
+		sort.Strings(dupes)
+
+		for _, name := range dupes {
+			pkgVersions := packages[name]
 
-		fmt.Printf("package %s imported as:\n", name)
-		for hash, imp := range pkgVersions {
-			fmt.Printf("  - %s %s\n", imp.version, hash)
-			for _, p := range imp.parents {
-				fmt.Printf("    - %s\n", p)
+			hashes := make([]string, 0, len(pkgVersions))
+			for h := range pkgVersions {
+				hashes = append(hashes, h)
+			}
+
+			sort.Slice(hashes, func(i, j int) bool {
+				ih := hashes[i]
+				jh := hashes[j]
+				iv := pkgVersions[ih].version
+				jv := pkgVersions[jh].version
+				switch iv.Compare(jv) {
```

Comment:
This can be written as:
```go
if res := iv.Compare(jv); res != 0 {
    return res
} else {
    return ih < jh
}
```

It is up to you.

Reply:
Good point (although I believe it would be `res < 0`).

Reply:
Right.